pkg/render/logstorage.go

```diff
@@ -522,22 +526,33 @@ func (es elasticsearchComponent) elasticsearchCluster() *esv1.Elasticsearch {
 			NodeSets: []esv1.NodeSet{
 				{
 					Count: int32(nodeConfig.Count),
-					Name:  "es",
+					Name: nodeSetName(pvcTemplate),
 					Config: &cmnv1.Config{
 						Data: map[string]interface{}{
 							"node.master": "true",
 							"node.data":   "true",
 							"node.ingest": "true",
 						},
 					},
-					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{es.pvcTemplate()},
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{pvcTemplate},
 					PodTemplate:          es.podTemplate(),
 				},
 			},
 		},
 	}
 }
 
+// nodeSetName returns thumbprint of PersistentVolumeClaim object as string.
+// As storage requirements of NodeSets are immutable,
+// renaming a NodeSet automatically creates a new StatefulSet with new PersistentVolumeClaim.
+// https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-orchestration.html#k8s-orchestration-limitations
+func nodeSetName(pvcTemplate corev1.PersistentVolumeClaim) string{
+	pvcTemplateHash := md5.New()
+	templateBytes, _ := json.Marshal(pvcTemplate)
+	pvcTemplateHash.Write(templateBytes)
+	// Slice to fit NodeSets Name's max length of 23 char
+	return hex.EncodeToString(pvcTemplateHash.Sum(nil))[:23]
```

Comment:
Everything hinges on this returning a unique hash. What guarantees the first 20 chars are unique?

Reply:
128bit is the minimum length of hash output, that would be 32 char. I was looking at options to get 23 char unique value, but all the docs I referred suggested truncating as collision will be pretty low even with that.

Reply:
> Everything hinges on this returning a unique hash. What guarantees the first 20 chars are unique?
It would be _extremely_ unlikely that even the first 5 characters would be the same, using 23 characters is quite the overkill. One character differences in two strings going through a hash function usually generate wildly different hashes.

This will generate huge pod names, which might seem a little weird to our customers, and I think the likely hood of collision with even 5 suffixes is so small.

One other note is that only one suffix ever exists at a time, and we don't need to generate a hash that's never been used before, we only need to generate a hash that was different from the previous ones. The likely hood of generating two hash in a row with 5 character likes that have different generated from unique PVCs is extremely unlikely, to say the least.

Reply:
I've switched to non-cryptographic hash with smaller output length.